src/pages/workspaces/workspace/WorkspaceContainer.js

```diff
@@ -176,6 +176,36 @@ const useClusterPolling = namespace => {
   return { clusters, refreshClusters }
 }
 
+const useAppPolling = namespace => {
```

Comment:
Similarly don't like that this is duplicated from cluster code, but don't want to expect them to be identical forever.

Reply:
This one is interesting. There's potentially an abstraction here, along the lines of 'poll some resource, with variable interval, and return the data and a way to request a refresh'. It would have a fair number of parameters, though, so I'm not sure how nice it would look.

Reply:
Yeah, this is the one that seems interesting. I know we changed useClusterPolling to include persistent disks in the PD work, so theoretically we could fold this in there as well, but while Clusters and PDs seem closely related, apps feel like their own resource and so probably shouldn't refresh or poll in the same loop?

Reply:
Yep, I agree that the polling loops should be independent since the resources are.